packages/common/core/src/form/method.controller.ts

```diff
@@ -0,0 +1,66 @@
+import type { OdsCommonFieldAttribute, OdsCommonFieldValidityState } from './interfaces/attributes';
+import type { OdsCommonFieldMethod } from './interfaces/methods';
+import type { OdsCommonFieldEvent } from './interfaces/events';
+import { OdsInputValue } from './ods-input-value';
+
+class OdsCommonFieldMethodController<T extends OdsCommonFieldAttribute & OdsCommonFieldEvent> implements OdsCommonFieldMethod {
+
+  constructor(protected readonly component: T) { }
+
+  /**
+   * empty the value
+   */
+  async clear(): Promise<void> {
+    if (this.component.disabled) {
+      return;
+    }
+    this.component.value = null;
+    this.component.odsClear?.emit();
+  }
+
+  async getValidity<T extends { validity: ValidityState }>(element: T): Promise<OdsCommonFieldValidityState> {
+    const forbiddenValue = this.hasForbiddenValue();
+    const isValid = forbiddenValue ? false : element.validity.valid;
+    return {
+      ...element.validity,
```

Comment:
this won't work as ValidityState values seems to not be in the ob ject but in its `prototype` (so things like `Object.keys` for example won't work also).
You have to manually loop using `for .. in` to read the actual value adn generate a copy object

Reply:
Not like the `for..in` because of the ts-ignore & the cast of the type (`as unknown as ValidityState`). So I create a mapper